Source/AliveLibAE/Input.hpp

```diff
@@ -1,9 +1,150 @@
 #pragma once
 
 #include "../relive_lib/Function.hpp"
+#include "Types.hpp"
 
 typedef s32 (*t_InputCallback)();
 
+
+enum VK : u32
```

Comment:
That will slightly alter the behaviour, as with vanilla you could use each Shift/Alt/Ctrl interchangibly, (eg. run with either Shift), and binding it with SDL_KeyCode will force binding to specific Left or Right variant, as SDL_KeyCode/SDL_ScanCode don't have ambiguous Shift/Alt/Ctrl as a separate keycode for some reason (unrelated: though it does have them in SDL_Keymod).

That would also require some changes to the way sInputKeyStates_BD2F60[256], sAllowedGameKeys_5C9D30[256] work, cause the array length is tied to the enum value range of the VK, and e.g SDL_KeyCode's LSHIFT is of value 
"SDL_SCANCODE_LALT | (1 << 30)" which is equal to 1073742050

Alternatively, instead of hard-bounding our inputs to SDL_KeyCode we could bind them to SDL_ScanCode, but those span up to 512 - so we'd have to extend out structures or trim out the scancodes after 255 (which don't seem like something anyone would use anyway)

I'm not sure if using SDL_ScanCodes here is a proper way to do it, have mixed feelings as I'm not fully sure how they differ from KeyCodes. Apparently https://stackoverflow.com/questions/31941446/what-is-the-difference-between-an-sdl-physical-key-code-and-an-sdl-virtual-key-c are post OS-keyboard-layout . Then again, we used ScanCodes until now anyway, just in Sys.cpp

Reply:
InputCommands::eRun is bound to VK::eShift. There's; no SDL_KeyCode for Shift, only LShift or RShift, same with SDL_ScanCode.

Both current and previous VK implementation set the VK key to VK::Shift whenever LShift or RShift were pressed.